azurelinuxagent/common/protocol/wire.py

```diff
@@ -857,100 +857,77 @@ def check_wire_protocol_version(self):
-            logger.periodic_info(logger.EVERY_SIX_HOURS, msg)
+            logger.info(msg)
 
             try:
                 ret = host_func()
-                if ret:
-                    msg = "[PERIODIC] Request succeeded using the host plugin channel after goal state refresh. " \
+                if ret is not None:
+                    msg = "Request succeeded using the host plugin channel after goal state refresh. " \
                           "ContainerId changed from {0} to {1}, " \
-                          "role config file changed from {2} to {3}.".format(old_container_id, new_container_id,
-                                                                             old_role_config_name, new_role_config_name)
-                    add_periodic(delta=logger.EVERY_SIX_HOURS,
-                                 name=AGENT_NAME,
-                                 version=CURRENT_VERSION,
-                                 op=WALAEventOperation.HostPlugin,
-                                 is_success=True,
-                                 message=msg,
-                                 log_event=True)
+                          "role config file changed from {2} to {3}.".format(old_container_id, new_container_id, old_role_config_name, new_role_config_name)
+                    add_event(AGENT_NAME, op=WALAEventOperation.HostPlugin, version=CURRENT_VERSION, is_success=True, message=msg, log_event=True)
 
-            except (ResourceGoneError, InvalidContainerError) as e:  # pylint: disable=C0103
-                msg = "[PERIODIC] Request failed using the host plugin channel after goal state refresh. " \
+            except Exception as error:
+                msg = "Request failed using the host plugin channel after goal state refresh. " \
                       "ContainerId changed from {0} to {1}, role config file changed from {2} to {3}. " \
-                      "Exception type: {4}.".format(old_container_id, new_container_id, old_role_config_name,
-                                                    new_role_config_name, type(e).__name__)
-                add_periodic(delta=logger.EVERY_SIX_HOURS,
-                             name=AGENT_NAME,
-                             version=CURRENT_VERSION,
-                             op=WALAEventOperation.HostPlugin,
-                             is_success=False,
-                             message=msg,
-                             log_event=True)
-                raise
+                      "Exception type: {4}.".format(old_container_id, new_container_id, old_role_config_name, new_role_config_name, type(error).__name__)
+                add_event(AGENT_NAME, op=WALAEventOperation.HostPlugin, version=CURRENT_VERSION, is_success=False, message=msg, log_event=True)
+                return None
+        except Exception as error:
+            logger.info("Request failed using the host plugin channel. Error: {0}", ustr(error))
+            return None
```

Comment:
we also changing the way we are reporting errors here... previously we were raising and now we are returning None.

this changes the interface of upload_logs

Reply:
Upload logs doesn't use this function. It uses a method from `hostplugin.py`.

Reply:
It actually uses this function, you were right. The code has been updated, the interface of upload_logs remains the same as before.